Spokestack/AppleSpeechRecognizer.swift

```diff
@@ -83,15 +84,18 @@ import Speech
     }
     
     private func deactivate() {
-        self.active = false
-        self.context.isActive = false
-        self.recognitionTask?.finish()
-        self.recognitionTask = nil
-        self.recognitionRequest?.endAudio()
-        self.recognitionRequest = nil
-        self.wakeActiveMaxWorker?.cancel()
-        self.audioEngine.pause()
-        self.context.dispatch(.deactivate)
+        if self.active {
+            self.active = false
+            self.context.isActive = false
+            self.recognitionTask?.finish()
+            self.recognitionTask = nil
+            self.recognitionRequest?.endAudio()
+            self.recognitionRequest = nil
+            self.vadFallWorker?.cancel()
+            self.wakeActiveMaxWorker?.cancel()
+            self.audioEngine.pause()
+            self.context.dispatch(.deactivate)
```

Comment:
This is a good reason for event dispatching to happen in the speech context. If I'm reading correctly, all calls other than the event dispatch (and maybe the audio engine pause?) should be idempotent, and if the context managed dispatching events on activity edge, you'd have nothing to worry about here.